agent-bootstrap/src/main/java/io/opentelemetry/auto/typedspan/HttpServerSpan.java

```diff
@@ -0,0 +1,689 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.opentelemetry.auto.typedspan;
+
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.Tracer;
+import java.util.logging.Logger;
+
+/**
+ * <b>Required attributes:</b>
+ *
+ * <ul>
+ *   <li>http.method: HTTP request method.
+ * </ul>
+ *
+ * <b>Conditional attributes:</b>
+ *
+ * <ul>
+ *   <li>http.status_code: [HTTP response status
+ *       code](https://tools.ietf.org/html/rfc7231#section-6).
+ * </ul>
+ *
+ * <b>Additional constraints</b>
+ *
+ * <p>At least one of the following must be set:
+ *
+ * <ul>
+ *   <li>http.url
+ *   <li>http.scheme, http.host, http.target
+ *   <li>http.scheme, http.server_name, net.host.port, http.target
+ *   <li>http.scheme, net.host.name, net.host.port, http.target
+ * </ul>
+ */
+public class HttpServerSpan extends DelegatingSpan implements HttpServerSemanticConvention {
```

Comment:
Can this extend `HttpSpan`?

Reply:
At the moment no because `HttpServerSpan.AttributeStatus` is not compatible with `HttpSpan.AttributeStatus`.